src/main/java/com/pinterest/secor/io/impl/AvroParquetFileReaderWriterFactory.java

```diff
@@ -66,6 +73,28 @@ public FileWriter BuildFileWriter(LogFilePath logFilePath, CompressionCodec code
         return serialized.array();
     }
 
+    protected static GenericRecord deserializeAvroRecord(DecoderFactory decoderFactory, SpecificDatumReader<GenericRecord> reader, byte[] value) throws IOException {
+        BinaryDecoder binaryDecoder = decoderFactory.binaryDecoder(value, null);
+        return reader.read(null, binaryDecoder);
+    }
+
+    protected GenericRecord decodeMessage(byte[] value, String topic, SpecificDatumReader<GenericRecord> reader) throws IOException {
+        if (value.length > 1 && value[0] == 0) {
```

Comment:
Where does this convention coming from (use the first byte to differentiate)?

Reply:
I added a comment in the code:
Avro schema registry header format is a "Magic Byte" that equals 0 followed by a 4-byte int